backend/models/QnA_Relevance.js

```diff
@@ -0,0 +1,59 @@
+import natural from 'natural'
+const TfIdf = natural.TfIdf
+
+
+export function dotProduct(a, b) {
+    let product = 0;
+    for (let i = 0; i < a.length; i++) {
+        product += a[i] * b[i];
+    }
+    return product;
+}
+
+export function magnitude(vector) {
+    let sum = 0;
+    for (let i = 0; i < vector.length; i++) {
+        sum += vector[i] ** 2;
+    }
+    return Math.sqrt(sum);
+}
+
+export function cosineSimilarity(a, b) {
+    const dotProd = dotProduct(a, b);
+    const magA = magnitude(a);
+    const magB = magnitude(b);
+
+    if (magA === 0 || magB === 0) {
+        return 0;
+    }
+
+    const cosSim = dotProd / (magA * magB);
+    return cosSim;
+}
+
+export function getRelevanceScore(question, response){
```

Comment:
would be great to have a link to diagram or comment explaining logic for the score/how it's calculated!